lang/cs/Org.Apache.REEF.IMRU/OnREEF/ResultHandler/WriteResultHandler.cs

```diff
@@ -83,7 +82,7 @@ public void Dispose()
 
             if (_fileSystem.Exists(remoteUri))
             {
-                Exceptions.Throw(
+                Utilities.Diagnostics.Exceptions.Throw(
```

Comment:
I added Org.Apache.REEF.IMRU.OnREEF.Exceptions.TaskStateTransitionException based on the review comment. After that, all Exceptions.Throw which actually refer to Utilities.Diagnostics. shows red color in the code and cannot compile. 

I am pretty sure we will need to add more exceptions in Org.Apache.REEF.IMRU.OnREEF.Exceptions when we move on for Fault tolerant work. We have to specify name space some where to distinct those two. 

This happens already in many other projects in REEF. An example is REEF.Common.Exceptions. In other classes in REEF.Common where Utilities.Diagnostics.Exception,  we have to use namespace. 

Example is EvaluatorRuntime.cs, you can find a lot of usage like:
Utilities.Diagnostics.Exceptions.CaughtAndThrow